### ISSUE TITLE
Null testValue causing exception

### ISSUE THREAD

**Issue by CalebFenton** [j256/simplemagic#18]
Hello,

I'm getting an NPE when using magic files installed after making and installing `file` from source. Here's how I construct the `ContentInfoUtil`:

```java
ContentInfoUtil util = new ContentInfoUtil("/usr/local/share/misc/magic");
```

This is the error it generates:

```
java.lang.NullPointerException
	at com.j256.simplemagic.types.StringType.getStartingBytes(StringType.java:97)
	at com.j256.simplemagic.entries.MagicEntry.getStartsWithByte(MagicEntry.java:96)
	at com.j256.simplemagic.entries.MagicEntries.optimizeFirstBytes(MagicEntries.java:90)
	at com.j256.simplemagic.ContentInfoUtil.readEntriesFromFile(ContentInfoUtil.java:341)
	at com.j256.simplemagic.ContentInfoUtil.<init>(ContentInfoUtil.java:156)
	at com.j256.simplemagic.ContentInfoUtil.<init>(ContentInfoUtil.java:103)
```

The issue seems to be that `testValue` is null on this line:

```java
return ((TestInfo) testValue).getStartingBytes();
```

This happens because a lot of rules have an 'x' for the test value. The `testValue` is set to `null` in MagicEntryParser:

```java
// process the test-string
Object testValue;
String testStr = parts[2];
if (testStr.equals("x")) {
	testValue = null;
```

I admit to not fully auditing the code. I just wanted to run it by you to make sure my problem isn't something stupid that I'm doing. I figured the official file magic should work but there are a lot of entries with null test values. Any ideas? Thanks.

**Comment by j256** [comment]
Huh.  This looks like a standard bug in the code @CalebFenton .   The patterns may be invalid since they are doing a test but not specifying a test-value but the code should be resistant to that.

**Comment by CalebFenton** [comment]
Interesting. Thanks for looking into this so quickly.

FYI, an example of a problem line is in `compress`:

```
# Zlib https://www.ietf.org/rfc/rfc6713.txt
0	string/b	x
>0	beshort%31	=0
>>0	byte&0xf	=8
>>>0	byte&0x80 	=0	zlib compressed data
!:mime	application/zlib
```

If I comment out `0	string/b	x`, everything works fine. This is consistent with what you say the problem is.

**Comment by CalebFenton** [comment]
Apparently, that is the _only_ line that causes a problem in the entire magic directory. It's possible to work around the issue by simply commenting the entire  zlib entry.

**Comment by j256** [comment]
Well the right thing to do is to fix the code and then to fix the entry if it is even in error.

**Comment by j256** [comment]
This has been fixed and will be in version 1.10.
